_includes/configuration/formats.md

```diff
@@ -42,18 +44,320 @@ Some built-in formats `fontsize`, `fontname`, `forecolor`, `hilitecolor` uses a
+      { selector: 'span', classes: 'red', remove: 'empty' },
+
+      // Configures `clear formatting` to remove the class green from spans and if the element then becomes empty it's left intact
+      { selector: 'span', classes: 'green', remove: 'none' }
+    ]
+  }
+});
+```
+
+#### split
+
+Enables you to control if partially selected inline elements should be split or not when you remove the format.
+
+**Type:** `boolean`
+
+**Default:** true for `inline` formats
+
+##### Example
+
+```js
+tinymce.init({
+  selector: 'textarea',
+  formats: {
+    bold: [
+      // Configures the default bold format to never split bold elements
+      { inline: 'strong', split: false }
+    ]
+  }
+});
+```
+
+#### expand
+
+Expand lets you specify if formats should be removed on matching parent elements when the selection is collapsed. So if you toggle of a format while having just the caret inside a element maching that specific format would remove the format. If this is set to false then it would only be removed it the selection is expanded over that specific format.
+
+**Type:** `boolean`
+
+##### Example
+
+```js
+tinymce.init({
+  selector: 'textarea',
+  style_formats: [
+    // A custom format that that expands to the closes parent element if the selection is collapsed
+    { title: 'My custom format', selector: 'div', classes: 'myclass', expand: true }
+  ]
+});
+```
+
+#### block_expand
+
+This option lets you control if the selection should expand upwards to the closest matching block element. This can be useful when configuring removeformat to remove block elements. So if the selection start is at the start of a matching block then that matching block will be included as well. If the end of the selection is at the end of a matching block element then that parent element will be included as well.
+
+So if the selection is from a to b in this html contents `<h1><b>[a</b></h1><p>b]</p>` then the h1 will be removed even if it's not part of the actual selection.
+
+**Type:** `boolean`
+
+##### Example
+
+```js
+tinymce.init({
+  selector: 'textarea',
+  formats: {
+    removeformat: [
+      {
+        selector: 'h1,h2,h3,h4,h5,h6',
+        remove: 'all',
+        split: false,
+        expand: false,
+        block_expand: true,
+        deep: true
+      },
+      {
+        selector: 'a,b,strong,em,i,font,u,strike,sub,sup,dfn,code,samp,kbd,var,cite,mark,q,del,ins',
+        remove: 'all',
+        split: true,
+        expand: false,
+        deep: true
+      },
+      { selector: 'span', attributes: ['style', 'class'], remove: 'empty', split: true, expand: false, deep: true },
+      { selector: '*', attributes: ['style', 'class'], split: false, expand: false, deep: true }
+    ]
+  }
+});
+```
+
+
+
+#### deep
+
+Enables you to control if formats of children with the same matching format should be removed as well. This is set to true by default on selector formats since if you select a table
```

Comment:
@spocke should this instead say "is set to false by default" (and the same below)? As looking at https://github.com/tinymce/tinymce/blob/master/src/core/main/ts/fmt/FormatRegistry.ts#L29 it mentions it's false by default and doing a quick test by registering a format it was false when using a selector.